aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -1096,6 +1096,119 @@ Tensor clone(const Tensor& src, c10::optional<c10::MemoryFormat> optional_memory
   return self;
 }
 
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ covariations ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tensor cov(const Tensor& self, 
+           const c10::optional<Tensor>& otherOpt,
+           c10::optional<bool> rowvarOpt, 
+           c10::optional<bool> biasOpt, 
+           c10::optional<int64_t> ddofOpt,
+           const c10::optional<Tensor>& fweightsOpt, 
+           const c10::optional<Tensor>& aweightsOpt) {
+  
+  const Tensor other = otherOpt.value_or(Tensor());
+  const Tensor fweights = fweightsOpt.value_or(Tensor());
+  const Tensor aweights = aweightsOpt.value_or(Tensor());
+  auto bias = biasOpt.value_or(false);
+  auto ddof = ddofOpt.value_or(0);
+  auto rowvar = rowvarOpt.value_or(true);
+
+  if(self.size(0) == 0)
+    return at::empty_like(self);  
+  TORCH_CHECK(self.dim() <= 2,
+              "Input1 has more than 2 dimensions." );
+    
+  Tensor X, y;
+  if(self.dim() == 1)
+    X = self.unsqueeze(0);
+  else
+    X = self;
+
+  if(!rowvar && X.size(0) != 1)
+    X = X.t();
+
+  if(other.defined()) {
+    TORCH_CHECK(other.dim() <= 2,
+                "Input2 has more than 2 dimensions." );
+ 
+    if(other.dim() == 1)
+      y = other.unsqueeze(0);
+    else
+      y = other;
+
+    if(!rowvar && y.size(0) != 1)
+      y = y.t();
+
+    X = at::cat({X, y}, 0);
+  }
+
+  if (ddofOpt == c10::nullopt)
```

Comment:
This is odd because the value of ddof was already set once. It should only be set in one place (either here or at the start of the function).